projects/plugins/jetpack/tests/php/sync/test_class.jetpack-sync-posts.php

```diff
@@ -882,7 +882,14 @@ function test_remove_related_posts_from_filtered_content() {
 
 		// Make sure that the related posts show up.
 		add_filter( 'jetpack_relatedposts_filter_enabled_for_request', '__return_true', 99999 );
-		Jetpack_RelatedPosts::init()->action_frontend_init();
+
+		$jetpack_relatedposts_mock = $this->getMockBuilder( 'Jetpack_RelatedPosts' )
+									->setMethods( array( 'is_fse_theme' ) )
+									->getMock();
+		$jetpack_relatedposts_mock
+			->method( 'is_fse_theme' )
+			->willReturn( false );
+		$jetpack_relatedposts_mock->action_frontend_init();
```

Comment:
Since `Jetpack_RelatedPosts` no longer has an `is_fse_theme` method, and the `Blocks::is_fse_theme` method can't be mocked directly because it's a static method, I wonder if we can add a filter to the `Blocks::is_fse_theme` method in a similar approach to `Blocks::is_amp_request()` and `Blocks::is_standalone_block()`. Then in the test, you can add a filter to return the value you expect? That's the approach I used in https://github.com/Automattic/jetpack/pull/19197/files in case it helps.

Though from looking at this test, I don't quite understand why it needs to be mocked, since I would have thought that `Blocks::is_fse_theme()` should be returning false by default? Unless the build in CI running the function that calls `Blocks::is_fse_theme()` doesn't include the update to the Blocks package, in which case attempting to call that function could be erroring out earlier, which could result in the related blocks content not being appended. I haven't double-checked any of that, though, so hopefully it isn't too much of a rabbit hole!

Reply:
Thank you for the filter suggestion! That has it working. I also don't quite understand why it needs to be mocked, since `Blocks::is_fse_theme()` is false by default in the Blocks package tests.

Reply:
Glad that got it working! It is a little bemusing, though!